File: backend/app/api/v1/history.py
from fastapi import APIRouter, Query
from app.database import supabase

router = APIRouter(prefix="/history", tags=["history"])


@router.get("")
def get_history(event_type: str = Query(None), limit: int = Query(20, le=50)):
    try:
        query = supabase.table("historical_events").select("*").order("date", desc=True)
        if event_type:
            query = query.eq("event_type", event_type.upper())
        result = query.limit(limit).execute()
        return {"events": result.data or []}
    except Exception as e:
        return {"error": str(e)}

File: frontend/src/api/history.ts
import { get } from './client'
import type { HistoricalEvent } from '../types/history'

interface HistoryResponse {
  events: HistoricalEvent[]
}

export function fetchHistory(event_type?: string): Promise<HistoryResponse> {
  const query = event_type ? `?event_type=${event_type}` : ''
  return get<HistoryResponse>(`/history${query}`)
}
